examples/graph_animations/example1.md

```diff
@@ -0,0 +1,193 @@
+## Graph traversal using BFS
+
+### Points covered
+1. Graph creation
+2. Algorithm explanation through two different visualisations
+
+### Graph Creation
+
+The graph object can be created/initialized by the use of a LightGraph typed object or any arbitrary graph types. In the latter case, there is a need to specify some additional accessibility functions to make use of some advanced visualisation features. It is discussed in later examples. This one covers how to do it with the use of a simple adacency list. It is supposed to be the most simplest way one can use the API with almost a zero learning curve.
+
+**Graph representation**
+```julia
+graph = [[2, 3, 4, 5],
+         [6, 7],
+         [8],
+         [],
+         [],
+         [],
+         [],
+         []]
+```
+The input graph can be of any Julia data type. This does not impose any restriction on the type of graph but for advanced features it requires some extra work. An alternative to this is using `LightGraphs.jl` for which there are many convenience functions available. Its usage is covered in [Example 2](example2.md).
+
+**Graph initialization**
+```julia
+# Parameters - Graph object | is directed? | width | height | starting position
+ga = GraphAnimation(graph, true, 300, 300, O)
+```
+
+**Node registration**
+```julia
+nodes = [Object(@Frames(prev_start()+5, stop=100), GraphNode(i, drawNode; animate_on=:scale, fill_color="yellow", border_color="black", text=string(i), text_valign=:middle, text_halign=:center)) for i in range(1, 8; step=1)]
```

Comment:
It's not clear how this is linked to `ga`

Reply:
In general what does `ga` do if I need to create the graph nodes and edges myself?

Reply:
I thought of keeping a reference internally to the currently active graph, like the way there is a  `CURRENT_VIDEO`. This however is not completely free of problems and restricts the way nodes can be added to graphs. I can change it to `GraphNode(ga, i, drawNode); ...`.

When adding a node, the first important thing it needs to do is generate a new layout. This however, is not done if the `mode` is `static` in which case it is generated only once when the whole layout is known.
The second thing, an ordering is kept in the graph object as to which node/edge appeared first. This helps in the utility functions like `animate_neighbors` where instead of placing actions on all neighboring nodes in the same frame range this ordering can be used to organize them sequentially.

Reply:
I'm just wondering what `ga` does with the `graph` object itself when one has to add the edges and nodes later anyway and what would happen if I add more nodes than in my `graph` ?

Reply:
And yeah I prefer `GraphNode(ga, i, drawNode` and optionally remove the `i` when one adds them in order anyway.